test/integration/StreamrClient.test.js

```diff
@@ -110,6 +112,136 @@ describe('StreamrClient Connection', () => {
         done()
     })
 
+    describe('resend', () => {
+        let client
+        let stream
+
+        let timestamps = []
+
+        beforeEach(async () => {
+            client = createClient()
+            await client.ensureConnected()
+
+            stream = await client.createStream({
+                name: uniqueId(),
+            })
+
+            timestamps = []
+            for (let i = 0; i < 5; i++) {
+                const message = {
+                    msg: `message${i}`,
+                }
+
+                // eslint-disable-next-line no-await-in-loop
+                const rawMessage = await client.publish(stream.id, message)
+                timestamps.push(rawMessage.getStreamMessage().getTimestamp())
+            }
+
+            await wait(2000) // wait for messages to (probably) land in storage
+        })
+
+        afterEach(async () => {
+            await client.disconnect()
+        })
+
+        it('resend last', async (done) => {
+            const messages = []
+
+            const sub = await client.resend(
+                stream.id,
+                {
+                    resend: {
+                        last: 3,
+                    },
+                },
+                (message) => {
+                    messages.push(message)
+                },
+            )
+
+            sub.on('resent', () => {
```

Comment:
for good measure, these listeners should probably all be `sub.once(`

Reply:
👍